host/bcm283x/gpio.go

```diff
@@ -1216,14 +1277,24 @@ func (d *driverGPIO) Init() (bool, error) {
 	if !Present() {
 		return false, errors.New("bcm283x CPU not detected")
 	}
-	model := distro.CPUInfo()["model name"]
-	if strings.Contains(model, "ARMv6") {
+	dTCompatible := strings.Join(distro.DTCompatible(), " ")
+	if strings.Contains(dTCompatible, "bcm2708") {
+		// RPi0/1.
 		d.baseAddr = 0x20000000
 		d.dramBus = 0x40000000
-	} else {
+		UseLegacyPull = true
+	} else if strings.Contains(dTCompatible, "bcm2709") {
 		// RPi2+
 		d.baseAddr = 0x3F000000
 		d.dramBus = 0xC0000000
+		UseLegacyPull = true
+	} else {
```

Comment:
Raspberry Pi 3 shows up as bcm2837.
I'll have to boot up my RPi1 and RPiZ to confirm.

Reply:
I assume it would be ok if nothing changed after #25 ?

Reply:
Humm it's weird, because in the issue Torsten noted that bcm2709 is returned on a RPi3 Raspbian Jessie but it's definitely bcm2837 when run locally on Raspbian Buster.

I'd prefer to keep add the new check for RPi4 first, then fallback on the previous check. This feels safer, unless you have another idea.

Reply:
If all we need to worry about are Raspberry Pis (Seems legit as some sources state that those chips are made specifically for RPi.), we can parse revision code found in `/proc/cpuinfo`. Otherwise, I think we can workaround this by detecting both naming schemes.

Reply:
I think your implementation is fine.